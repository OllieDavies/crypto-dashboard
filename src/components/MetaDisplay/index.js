import React from "react";

const MetaDisplay = ({ name, value, children }) => (
  <div>
    <p>
      <strong>{name}:</strong> {value || children}
    </p>
  </div>
);

export default MetaDisplay;
