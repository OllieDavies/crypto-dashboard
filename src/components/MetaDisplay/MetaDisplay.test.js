import React from "react";
import { render } from "@testing-library/react";
import MetaDisplay from ".";

describe("<MetaDisplay />", () => {
  it("matches snapshot with name and value props", () => {
    const { container } = render(<MetaDisplay name={"Name"} value={"Value"} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with name and children props", () => {
    const { container } = render(
      <MetaDisplay name={"Name"} children={<strong>Child</strong>} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
