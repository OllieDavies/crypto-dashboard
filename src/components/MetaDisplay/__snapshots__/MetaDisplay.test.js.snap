// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaDisplay /> matches snapshot with name and children props 1`] = `
<div>
  <p>
    <strong>
      Name
      :
    </strong>
     
    <strong>
      Child
    </strong>
  </p>
</div>
`;

exports[`<MetaDisplay /> matches snapshot with name and value props 1`] = `
<div>
  <p>
    <strong>
      Name
      :
    </strong>
     
    Value
  </p>
</div>
`;
